### DISCUSSION TITLE
11.1.0-rc1

### DISCUSSION THREAD

**Discussion by grokys** [AvaloniaUI/Avalonia#16026] [General]
This release contains bugfixes for issues reported for 11.1.0-beta2. If no showstoppers are reported this will become 11.1.0.

**Full Changelog**: https://github.com/AvaloniaUI/Avalonia/compare/11.1.0-beta2...11.1.0-rc1

<hr /><em>This discussion was created from the release <a href='https://github.com/AvaloniaUI/Avalonia/releases/tag/11.1.0-rc1'>11.1.0-rc1</a>.</em>

**Comment by StefanKoell** [comment]
I'm confused. I'm desperately waiting for this [PR](https://github.com/AvaloniaUI/Avalonia/pull/15517) to be released and it's still not in the 11.1.0-rc1?

**Reply by timunie** [reply]
it has the 11.1 backport label, which means will be in 11.1.1 if I understood correctly. For 11.1.0 it's too late I think.

**Reply by StefanKoell** [reply]
well, if that's not confusing... anyhow, thanks for the clarification. I hope it will be included then.

**Reply by grokys** [reply]
Yeah the 11.1 release has been trapped in feature creep hell for the last I don't know how many months. We're trying to get out of that now, so only showstopper bug fixes will be merged in the beta -> RC phase.

**Comment by viktorspacil** [comment]
Hi, I switch to RC1 from v11.0.11 and I encountered following "problems":
- I had to newly explicitly register DefaultValueConverter for bindings to automatically convert Methods to ICommand, is this intended?
- Custom converter specified on binding is newly called only once at init and not with every PropertyChange event (fired from ViewModel), is this intended?
- Avalonia.Xaml.Behaviors: I had EventTriggerBehavior with SourceObject="{Binding $self}" when EventTriggerBehavior element was nested directly within Window element for example. I had to switch to SourceObject="{Binding ElementName=AppWindow}" because syntax using $self stopped referencing the parent Window element (application crash), is this intended?
Am I missing something? Thank you.

**Reply by maxkatz6** [reply]
1., 2. No, sounds like a bug.
3. Pretty sure it is intended, and $self references current element and not the root. But I might be wrong.

**Reply by viktorspacil** [reply]
Do you want me to create issues with simple repro app for the 1 and 2 or are you able to simulate this behavior on your side? Thank you.

**Reply by grokys** [reply]
> I had to newly explicitly register DefaultValueConverter for bindings to automatically convert Methods to ICommand, is this intended?

Do you mean you're trying to do something like `<Button Command="{Binding Foo}"/>` where `Foo` is a method on your view model? That should definitely work: we have unit tests for it and I just tested it manually and it seems to work with both compiled and reflection bindings.

> Custom converter specified on binding is newly called only once at init and not with every PropertyChange event (fired from ViewModel), is this intended?

Do you mean a value converter specified in a binding, e.g. `<TextBlock Text="{Binding Bar, Converter={x:Static local:FooConverter.Instance}}"/` ? Again, this is something that I've just tried and didn't see any problems (and we also have unit tests for).

I suspect you might be hitting some edge cases in these issues, so yeah a repro would be helpful 🙏

**Reply by viktorspacil** [reply]
Bug #2 - DefaultValueConverter, below is slightly modified starting template of Avalonia MVVM project.

This is a working code (I had to newly and explicitly register and set the default value converter for the binding):
```
<Window
    Icon="/Assets/avalonia-logo.ico"
    Title="TestApp"
    d:DesignHeight="450"
    d:DesignWidth="800"
    mc:Ignorable="d"
    x:Class="TestApp.Views.MainWindow"
    x:DataType="vm:MainWindowViewModel"
    xmlns="https://github.com/avaloniaui"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:vm="using:TestApp.ViewModels"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">

    <Window.Styles>
        <Style Selector="Button.ClickMe">
            <Setter Property="Command" Value="{Binding TestMethod, Converter={StaticResource DefaultValueConverter}}" />
        </Style>
    </Window.Styles>

    <StackPanel Orientation="Vertical">
        <TextBlock
            HorizontalAlignment="Center"
            Text="{Binding Greeting}"
            VerticalAlignment="Center" />
        <Button
            Classes="ClickMe"
            Content="Click me"
            HorizontalAlignment="Center" />
    </StackPanel>

</Window>
```


This is not working (button is grey out/disabled, this code was working with v11.0.10 but is not working with the rc1):

```
<Window
    Icon="/Assets/avalonia-logo.ico"
    Title="TestApp"
    d:DesignHeight="450"
    d:DesignWidth="800"
    mc:Ignorable="d"
    x:Class="TestApp.Views.MainWindow"
    x:DataType="vm:MainWindowViewModel"
    xmlns="https://github.com/avaloniaui"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:vm="using:TestApp.ViewModels"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">

    <Window.Styles>
        <Style Selector="Button.ClickMe">
            <Setter Property="Command" Value="{Binding TestMethod}" />
        </Style>
    </Window.Styles>

    <StackPanel Orientation="Vertical">
        <TextBlock
            HorizontalAlignment="Center"
            Text="{Binding Greeting}"
            VerticalAlignment="Center" />
        <Button
            Classes="ClickMe"
            Content="Click me"
            HorizontalAlignment="Center" />
    </StackPanel>

</Window>
```

So it seems that the converters stopped working for binding which are set using styles.

**Reply by timunie** [reply]
please provide a minimal sample in a new issue. Need to be inspected deeper.

**Reply by viktorspacil** [reply]
I have created issue #16113.

**Reply by viktorspacil** [reply]
Hi, I also created an issue #16137 (with minimal sample) for the problem with the custom converter which is called only once at startup.

**Comment by dbriard** [comment]
Could you fix this issue with Text that has a width if 0 when TextAlignment is not Left in 11.1? https://github.com/AvaloniaUI/Avalonia/issues/16009
I am using HorizontalAlignment as a workaround but the rendering is not correct for multilines texts.

**Reply by grokys** [reply]
Not in the release candidate state, no, sorry. This stage is limited to showstopper bugs. A fix for that can likely go into 11.1.x though.

**Comment by cricketthomas** [comment]
Since upgrading from 11.1.0-beta2 to 11.1.0-rc1 I have noticed a small difference in text rendering when the background is set to Mica, is this expected? The text looks a bit thicker/more bold. It can be fixed by setting the text to `RenderOptions.TextRenderingMode="Antialias"`

11.1.0-beta2
![11.1.0-beta2](https://github.com/AvaloniaUI/Avalonia/assets/15821271/c2af15df-a072-403d-a812-8baadbb1812e)

11.1.0-rc1
![11.1.0-rc2](https://github.com/AvaloniaUI/Avalonia/assets/15821271/f508750d-6a64-45b6-be42-3eb179d981a8)

**Reply by Gillibald** [reply]
That is a known bug with layering

**Reply by Gillibald** [reply]
> That is a known bug with layering

Non opaque backgrounds are also not working well with sub pixel rendering

**Reply by LyleRolleman-plst** [reply]
Any ETA on a release?
